src/compas/utilities/itertools.py

```diff
@@ -185,10 +189,79 @@ def linspace(start, stop, num=50):
 
 
 def flatten(list_of_lists):
-    """Flatten one level of nesting"""
+    """Flatten one level of nesting.
```

Comment:
Since numpy's `flatten` doesn't flatten just one level, I think this name might be confusing.  So I would suggest either renaming this function or replacing it with something that behaves as `numpy.flatten` does.  Eg
```
def flatten(l):
    if not hasattr(l[0], '__len__'):
        return l
    return flatten(list(chain(*l)))

Reply:
i have the feeling this will break a lot of existing code. for example, in case flatten is used on a list of lists of points, the new version will basically return a list of floats, and that is not the case with the previous version.
```python
for point in list_of_polylines:
    # point is a point in the old version
    # point is a coordinate in the new version
```